proposals/restructure-KF-docs-proposal.md

```diff
@@ -0,0 +1,226 @@
+This is an initial proposal, to be iterated on, following the discussion in [#2293](https://github.com/kubeflow/website/issues/2293).
+
+### Kubeflow documentation structure
+
+- **About**
+  - Kubeflow
+  - Use Cases
+  - Contributing to Kubeflow
+  - Community
+  - Google Summer of Code
+  - Docs
+  - Style Guide for Kubeflow docs
+  
+- **Getting Started**
+  - Kubeflow Overview
+  - Installing Kubeflow
+  - Cloud Installation
+    - AWS
+    - Azure
+    - Google Cloud
+    - IBM Cloud
+  - Kubernetes installation
+    - Overview of Deployment on Existing Clusters
+    - Kubeflow Deployment with kfctl_k8s_istio
+    - Multi-user, auth-enabled Kubeflow with kfctl_existing_arrikto
+    - Multi-user, auth-enabled Kubeflow with kfctl_istio_dex
+  - Workstation Installation
+    - Kubeflow on Linux
+    - Kubeflow on macOS
+    - Kubeflow on Windows
+    - MiniKF
+    - Deploy using MiniKF on GCP
+    - Deploying with minikube on a single node
+    - Kubeflow on MicroK8s
+- **Components of Kubeflow**
+  - Central Dashboard
+    - Central Dashboard 
+    - Registration Flow
+  - Metadata
+  - Jupyter Notebooks
+  - Fairing 
+    - Overview of Kubeflow Fairing
+    - Install Kubeflow Fairing
+    - Configure Kubeflow Fairing
+    - Fairing on Azure
+    - Fairing on GCP
+    - Tutorials
+    - Reference
+  - Feature Store
+    - Introduction to Feast
+    - Getting started with Feast
+  - Frameworks for Training
+    - Chainer Training
+    - MPI Training
+    - MXNet Training
+    - PyTorch Training
+    - TensorFlow Training (TFJob)
+  - Hyperparameter Tuning
+    - Introduction to Katib
+    - Getting started with Katib
+    - Running an experiment
+    - Katib Configuration Overview
+    - Environment Variables for Katib Components
+  - Pipelines
+    - Pipelines
+  - Tools for Serving
+    - Overview
+    - KFServing
+    - Seldon Core Serving
+    - BentoML
+    - NVIDIA Triton Inference Server
+    - TensorFlow Serving
+    - TensorFlow Batch Prediction
+  - Multi-Tenancy in Kubeflow
+    - Introduction to Multi-user Isolation
+    - Design for Multi-user Isolation
+    - Getting Started with Multi-user Isolation
+  - Miscellaneous
+    - Nuclio functions
+- **Jupyter Notebooks**
+  - Overview of Jupyter Notebooks in Kubeflow
+  - Set Up Your Notebooks
+  - Create a Custom Jupyter Image
+  - Submit Kubernetes Resources
+  - Build a Docker Image on GCP
+  - Troubleshooting Guide
+- **Pipelines**
```

Comment:
Why do not move Jupyter Notebooks and Pipelines under /components section, since they are also Kubeflow components ?

Reply:
@kubeflow/wg-notebook-leads What do you think about the Jupyter Notebook section?
@Bobgy What about Kubeflow Pipelines ?

Reply:
Clearly, "jupyter notebooks" is a component of kubeflow, and should be under the "components of Kubeflow" heading?

But I thought we were going to have a separate versioned docs for components?

Reply:
I don't think it is necessarily settled that component applications will have their own sites, but it seems like it makes sense to allow each component to version their docs as necessary.

Reply:
@joeliedtke I agree that it seems that people want to separate the components into different domains. Anyhow those will still fit into the docs structure ([as pointed here](https://github.com/kubeflow/website/issues/2293#issuecomment-739049676)). 

Particularly for `notebooks` and `pipelines`, they will either fit at a top level in the tree (as today) or under `components`.

As @andreyvelich and @thesuperzapper mention, I think the second option makes sense. In addition, we can emphasize components by not using alphabetical order.

Reply:
Resolving this thread as it is outdated.